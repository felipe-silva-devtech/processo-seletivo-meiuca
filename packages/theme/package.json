{
  "name": "@psm/theme",
  "version": "1.0.0",
  "private": true,
  "main": "dist/cjs/index.js",
  "module": "dist/esm/index.js",
  "license": "MIT",
  "scripts": {
    "prebuild": "rm -rf dist",
    "build": "rollup -c"
  },
  "dependencies": {
    "@emotion/react": "^11.11.0"
  },
  "devDependencies": {
    "@rollup/plugin-commonjs": "^25.0.0",
    "@rollup/plugin-typescript": "^11.1.1",
    "rollup": "^2.79.1"
  }
}
