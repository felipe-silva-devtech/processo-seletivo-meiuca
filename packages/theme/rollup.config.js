import commonjs from "@rollup/plugin-commonjs";
import typescript from "@rollup/plugin-typescript";

const packageJson = require("./package.json");

export default {
  input: "src/index.ts",
  cache: false,
  output: [
    {
      file: packageJson.main,
      format: "cjs",
    },
    {
      file: packageJson.module,
      format: "esm",
    },
  ],
  plugins: [commonjs(), typescript({ tsconfig: "./tsconfig.json" })],
};
