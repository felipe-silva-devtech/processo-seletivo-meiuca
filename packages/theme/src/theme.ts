import tokens from "@psm/tokens";

export const theme = {
  lineHeights: {
    tight: tokens.lineHeightTight,
    medium: tokens.lineHeightMedium,
    distant: tokens.lineHeightDistant,
    superdistant: tokens.lineHeightSuperdistant,
  },
  fontSizes: {
    xxxs: tokens.fontSizeXxxs,
    xxs: tokens.fontSizeXxs,
    xs: tokens.fontSizeXs,
    sm: tokens.fontSizeSm,
    md: tokens.fontSizeMd,
    lg: tokens.fontSizeLg,
    xl: tokens.fontSizeXl,
    xxl: tokens.fontSizeXxl,
    xxxl: tokens.fontSizeXxxl,
    display: tokens.fontSizeDisplay,
    giant: tokens.fontSizeGiant,
  },
  radii: {
    none: tokens.borderRadiusNone,
    sm: tokens.borderRadiusSm,
    md: tokens.borderRadiusMd,
    lg: tokens.borderRadiusLg,
    pill: tokens.borderRadiusPill,
    circular: tokens.borderRadiusCircular,
  },
  borderWidths: {
    none: tokens.borderWidthNone,
    hairline: tokens.borderWidthHairline,
    thin: tokens.borderWidthThin,
    thick: tokens.borderWidthThick,
    heavy: tokens.borderWidthHeavy,
  },
  opacity: {
    semiopaque: tokens.opacityLevelSemiopaque,
    intense: tokens.opacityLevelIntense,
    medium: tokens.opacityLevelMedium,
    light: tokens.opacityLevelLight,
    semitransparent: tokens.opacityLevelSemitransparent,
  },
  shadows: {
    "01": tokens.shadowLevel1Stack,
    "02": tokens.shadowLevel2Stack,
    "03": tokens.shadowLevel3Stack,
    "04": tokens.shadowLevel4Stack,
  },
  space: {
    stack: {
      quarck: tokens.spacingStackQuarck,
      nano: tokens.spacingStackNano,
      xxxs: tokens.spacingStackXxxs,
      xxs: tokens.spacingStackXxs,
      xs: tokens.spacingStackXs,
      sm: tokens.spacingStackSm,
      md: tokens.spacingStackMd,
      lg: tokens.spacingStackLg,
      xl: tokens.spacingStackXl,
      xxl: tokens.spacingStackXxl,
      xxxl: tokens.spacingStackXxxl,
      huge: tokens.spacingStackHuge,
      giant: tokens.spacingStackGiant,
    },
    inset: {
      quarck: tokens.spacingInsetQuarck,
      nano: tokens.spacingInsetNano,
      xs: tokens.spacingInsetXs,
      sm: tokens.spacingInsetSm,
      md: tokens.spacingInsetMd,
      lg: tokens.spacingInsetLg,
    },
    squish: {
      quarck: tokens.spacingSquishQuarckStack,
      nano: tokens.spacingSquishNanoStack,
      xs: tokens.spacingSquishXsStack,
      sm: tokens.spacingSquishSmStack,
    },
    inline: {
      quarck: tokens.spacingInlineQuarck,
      nano: tokens.spacingInlineNano,
      xxxs: tokens.spacingInlineXxxs,
      xxs: tokens.spacingInlineXxs,
      xs: tokens.spacingInlineXs,
      sm: tokens.spacingInlineSm,
      md: tokens.spacingInlineMd,
      lg: tokens.spacingInlineLg,
      xl: tokens.spacingInlineXl,
    },
  },
  fonts: {
    highlight: tokens.fontFamilyHighlight,
  },
  fontWeights: {
    bold: tokens.fontWeightBold,
    medium: tokens.fontWeightMedium,
    regular: tokens.fontWeightRegular,
  },
  colors: {
    neutral: {
      "01": tokens.colorNeutral1,
      "02": tokens.colorNeutral2,
      "03": tokens.colorNeutral3,
      "04": tokens.colorNeutral4,
      "05": tokens.colorNeutral5,
    },
    brand: {
      primary: {
        "01": tokens.colorBrandPrimary1,
        "02": tokens.colorBrandPrimary2,
        "03": tokens.colorBrandPrimary3,
        "04": tokens.colorBrandPrimary4,
        "05": tokens.colorBrandPrimary5,
      },
    },
  },
};

export type Theme = typeof theme;
