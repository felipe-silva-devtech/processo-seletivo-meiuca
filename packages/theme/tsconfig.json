{
  "compilerOptions": {
    "lib": ["dom", "dom.iterable", "esnext"],
    "allowJs": true,
    "skipLibCheck": true,
    "strict": false,
    "forceConsistentCasingInFileNames": true,
    "noEmit": true,
    "incremental": true,
    "esModuleInterop": true,
    "module": "esnext",
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "jsx": "react",
    "declaration": true,
    "declarationDir": "dist"
  },
  "include": ["src/**/*.ts", "src/**/*.tsx"],
  "exclude": ["node_modules", "dist"]
}
