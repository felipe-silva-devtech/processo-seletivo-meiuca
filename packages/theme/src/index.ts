import { ThemeProvider } from "./provider";
import { theme, Theme } from "./theme";

export { ThemeProvider, theme };
export type { Theme };
