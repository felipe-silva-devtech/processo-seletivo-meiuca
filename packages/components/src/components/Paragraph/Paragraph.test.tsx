import React from "react";
import { render, RenderOptions } from "@testing-library/react";
import "@testing-library/jest-dom";

import { ThemeProvider, theme } from "@psm/theme";
import { Paragraph } from ".";

function renderWithWrapper(
  component?: React.ReactElement,
  options?: RenderOptions
) {
  const Wrapper = <ThemeProvider theme={theme}>{component}</ThemeProvider>;

  return render(Wrapper, options);
}

describe("Paragraph", () => {
  const content =
    "Lorem Ipsum is simply dummy text of the printing and typesetting industry";

  it("render component", () => {
    const { container, getByText } = renderWithWrapper(
      <Paragraph content={content} />
    );

    expect(getByText(content)).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  it("render component with children", () => {
    const { container, getByText } = renderWithWrapper(
      <Paragraph>{content}</Paragraph>
    );

    expect(getByText(content)).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
});
