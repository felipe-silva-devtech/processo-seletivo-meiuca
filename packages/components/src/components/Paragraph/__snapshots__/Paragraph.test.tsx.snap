// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph render component 1`] = `
<div>
  <p
    class="css-2wh3l1"
  >
    Lorem Ipsum is simply dummy text of the printing and typesetting industry
  </p>
</div>
`;

exports[`Paragraph render component with children 1`] = `
<div>
  <p
    class="css-2wh3l1"
  >
    Lorem Ipsum is simply dummy text of the printing and typesetting industry
  </p>
</div>
`;
